frontend/public/components/operator-lifecycle-manager/descriptors/spec/index.tsx

```diff
@@ -30,18 +34,53 @@ const ResourceRequirements: React.SFC<SpecCapabilityProps> = ({obj, descriptor})
   <dd><ResourceRequirementsModalLink type="requests" obj={obj} path={descriptor.path} /></dd>
 </dl>;
 
-const K8sResourceLink: React.SFC<SpecCapabilityProps> = (props) => <ResourceLink kind={props.capability.split(SpecCapability.k8sResourcePrefix)[1]} name={props.value} namespace={props.namespace} title={props.value} />;
+const K8sResourceLink: React.SFC<SpecCapabilityProps> = (props) => _.isEmpty(props.value)
+  ? <span className="text-muted">None</span>
+  : <ResourceLink kind={props.capability.split(SpecCapability.k8sResourcePrefix)[1]} name={props.value} namespace={props.namespace} title={props.value} />;
 
 const BasicSelector: React.SFC<SpecCapabilityProps> = ({value, capability}) => <Selector selector={value} kind={capability.split(SpecCapability.selector)[1]} />;
 
+class BooleanSwitch extends React.Component<SpecCapabilityProps, BooleanSwitchState> {
+  constructor(props) {
+    super(props);
+    this.state = {value: props.value, confirmed: false};
```

Comment:
In some new components, we've started declaring state as a class property without the constructor.

https://hackernoon.com/the-constructor-is-dead-long-live-the-constructor-c10871bea599